### PR TITLE
Personal/sowu/cli exception 2.0

### PR DESCRIPTION
**Pull request by sowu880**
## Description

Refine exception handling in tool.
throw exceptions to program.cs and output error message in program.cs.

### CODE REVIEW

**Review by qiwjin — APPROVED** [review]

File: src/Microsoft.Health.Fhir.Liquid.Converter.Tool/Program.cs [hunk 1]
```diff
@@ -14,19 +13,28 @@ namespace Microsoft.Health.Fhir.Liquid.Converter.Tool
...
+                parseResult.WithParsed<ConverterOptions>(options => ConverterLogicHandler.Convert(options));
+                await parseResult.WithParsedAsync<PullTemplateOptions>(options => TemplateManagementLogicHandler.PullAsync(options));
+                await parseResult.WithParsedAsync<PushTemplateOptions>(options => TemplateManagementLogicHandler.PushAsync(options));
+                parseResult.WithNotParsed((errors) => HandleOptionsParseError(parseResult));
+                return 0;
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine($"Process failed: {ex.Message}");
+                return -1;
```
Comment (qiwjin) [inline src/Microsoft.Health.Fhir.Liquid.Converter.Tool/Program.cs:30]: return 1?

Comment (sowu880) [reply src/Microsoft.Health.Fhir.Liquid.Converter.Tool/Program.cs:30]: @yankunhuang-pku What about return 1 when process failed?

**Comment by qiwjin** [comment]
please also help to add functional tests for it later.

File: src/Microsoft.Health.Fhir.Liquid.Converter.Tool/Program.cs [hunk 2]
```diff
@@ -14,19 +13,28 @@ namespace Microsoft.Health.Fhir.Liquid.Converter.Tool
...
+            }
         }
 
         private static void HandleOptionsParseError(ParserResult<object> parseResult)
         {
             var usageText = HelpText.RenderUsageText(parseResult);
-            Console.WriteLine(usageText);
+            throw new InputParameterException(usageText);
```
Comment (BoyaWu10) [inline src/Microsoft.Health.Fhir.Liquid.Converter.Tool/Program.cs:37]: Can we use `ArgumentException` instead of creating a new one?

**Review by BoyaWu10 — APPROVED** [review]

File: src/Microsoft.Health.Fhir.Liquid.Converter.Tool/Program.cs (hunk 1 above)
Comment (yankunhuang-pku) [reply src/Microsoft.Health.Fhir.Liquid.Converter.Tool/Program.cs:30]: As long as it is a non-zero number, I can identify it on my side.

**Review by yankunhuang-pku — COMMENTED** [review]
Is it expected that both error messages will be output at the same time? Should the program be terminated after an error occurs?
`Fail to pull templates: The output folder is not empty. If force to override, please add -f in parameters`
`Process Exits: Fail to pull templates. Error: failed to authorize: failed to fetch anonymous token: unexpected status: 401 Unauthorized`

**Review by yankunhuang-pku — APPROVED** [review]

**Pull request merged**